mongo/client_examples_test.go

```diff
@@ -314,3 +314,50 @@ func ExampleConnect_aWS() {
 	}
 	_ = ecClient
 }
+
+func ExampleConnect_versionedAPI() {
+	// Configure a client with versioned API.
+	//
+	// Versioned API is a new feature in MongoDB 5.0 that allows user-selectable API versions, subsets of MongoDB
+	// server semantics, to be declared on a client. During communication with a server, clients with a declared
+	// API version will force that server to behave in a manner compatible with the API version. Declaring an API
+	// version on your client can be used to ensure consistent responses from a server, providing long term API
+	// stability for an application.
+	//
+	// The declared API version is applied to all commands run through the client, including those sent through
+	// the generic command helper. Specifying versioned API options in the command document AND declaring an API
+	// version on the client is not supported and will lead to undefined behaviour. To run any command with a different
+	// API version or without declaring one, create a separate client that declares the appropriate API version.
+
+	// ServerAPIOptions must be declared with an API version. ServerAPIVersion1 is a constant equal to "1".
+	serverAPI := options.ServerAPI(options.ServerAPIVersion1)
+	serverAPIClient, err := mongo.Connect(context.TODO(), options.Client().SetServerAPIOptions(serverAPI))
+	if err != nil {
+		panic(err)
+	}
+	_ = serverAPIClient
+
+	// ServerAPIOptions can be declared with a Strict option. Declaring a strict API version will cause the MongoDB server
+	// to reject all commands that are not part of the declared API version. This includes command options and
+	// aggregation pipeline stages. For example, the following distinct call would fail because the distinct command is
+	// not part of API version 1:
+	serverAPIStrict := options.ServerAPI(options.ServerAPIVersion1).SetStrict(true)
+	serverAPIStrictClient, err := mongo.Connect(context.TODO(), options.Client().SetServerAPIOptions(serverAPIStrict))
+	if err != nil {
+		panic(err)
+	}
+
+	// Fails with error: (APIStrictError) Provided apiStrict:true, but the command distinct is not in API Version 1
+	_, err = serverAPIStrictClient.Database("db").Collection("coll").
```

Comment:
The multi-line `Distinct` call reads awkwardly. Might be better to declare a `Collection` on this line and then do `coll.Distinct()` on the next?